lib/keystores.js

```diff
@@ -119,37 +121,59 @@ api.insert = async ({config} = {}) => {
  *   retrieve.
  * @param {Object} [options.query={}] - The query to use.
  * @param {Object} [options.options={}] - The query options (eg: 'sort').
+ * @param {boolean} [options.explain=false] - An optional explain boolean.
  *
- * @return {Promise<Array>} resolves to the records that matched the query.
+ * @return {Promise<Array> | ExplainObject} resolves to the records that matched
+ *   the query or an ExplainObject if `explain=true`.
  */
-api.find = async ({
-  controller, query = {}, options = {}
+exports.find = async ({
+  controller, query = {}, options = {}, explain = false
 } = {}) => {
   assert.string(controller, 'options.controller');
+  const collection = database.collections['kms-keystore'];
+
   // force controller ID
   query['config.controller'] = controller;
-  return database.collections['kms-keystore'].find(query, options).toArray();
+
+  if(explain) {
+    const cursor = await collection.find(query, options);
+    return cursor.explain('executionStats');
+  }
+
+  return collection.find(query, options).toArray();
 };
 
 /**
  * Updates a keystore config if its sequence number is next.
  *
  * @param {Object} config the keystore configuration.
+ * @param {boolean} [options.explain=false] - An optional explain boolean.
  *
- * @return {Promise<Object>} resolves to the database record.
+ * @return {Promise<Object> | ExplainObject} resolves to the database record or
+ *   an ExplainObject if `explain=true`.
  */
-api.update = async ({config} = {}) => {
+exports.update = async ({config, explain = false} = {}) => {
   assert.object(config, 'config');
 
   // insert the configuration and get the updated record
   const now = Date.now();
```

Comment:
Noticing that this comment has not been correct (copy pasta from the insert function). Also, this `Date.now()` doesn't need to run in the `explain` case. So let's just call `Date.now()` directly via `'meta.updated': Date.now()` in the `updateOne` function instead of creating a separate var for it.

Reply:
This comment still needs to be addressed.